### ISSUE TITLE
[no-restricted-syntax] in recommended--extra's  should be it's own standalone rule

### ISSUE THREAD

**Issue by Samuel-Therrien-Beslogic** [angular-eslint/angular-eslint#1003]
**Description and reproduction of the issue**
`no-restricted-syntax` is a rule that is very likely to be overriden in project configurations. It is meant as a quick and easy way for everyone to add custom checks through AST selectors. Making the `console.log/warn/error (Unexpected property on console object was called)` rule in `recommended--extra` volatile and impractical.

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `13.2.1` |
| `@angular-eslint/template-parser`         | `13.2.1` |
| `@typescript-eslint/parser`               | `5.25.0` |
| `ESLint`                                  | `8.15.0` |
| `node`                                    | `16.13.2` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
 >  NX   Cannot find target 'version' for project 
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
v16 is now the latest release and the `recommended--extra` config no longer exists: https://github.com/angular-eslint/angular-eslint/releases/tag/v16.0.0

As with almost all other ESLint plugins, angular-eslint will focus on providing recommended configuration for its own rules. Please refer to typescript-eslint documentation and shared configs for the recommended setup for typescript-eslint rules:
https://typescript-eslint.io/linting/configs/#recommended-configurations
